tokio-trace/src/span.rs

```diff
@@ -1,6 +1,50 @@
-//! Spans represent periods of time in the execution of a program.
+//! Spans represent periods of time in which a program was executing in a
+//! particular context.
 //!
-//! # Entering a Span
+//! A span consists of [fields], user-defined key-value pairs of arbitrary data
+//! that describe the context the span represents, and [metadata], a fixed set
+//! of attributes that describe all `tokio-trace` spans and events. Each span is
+//! assigned an [`Id` ] by the subscriber that uniquely identifies it in relation
+//! to other spans.
+//!
+//! # Creating Spans
+//!
+//! Spans are created using the [`span!`] macro. This macro is invoked with a
+//! [verbosity level], followed by a set of attributes whose default values
+//! the user whishes to override, a string literal providing the span's name,
+//! and finally, between zero and 32 fields.
+//!
+//! For example:
+//! ```rust
+//! #[macro_use]
+//! extern crate tokio_trace;
+//! use tokio_trace::Level;
+//!
+//! # fn main() {
+//! /// Construct a new span at the `INFO` level named "my_span", with a single
+//! /// field named answer , with the value `42`.
+//! let my_span = span!(Level::INFO, "my_span", answer = 42);
+//! ```
+//!
+//! The documentation for the [`span!`] macro provides a complete set of
+//! examples for the various options that exist when creating spans.
+//!
+//! The [`trace_span!`], [`debug_span!`], [`info_span!`], [`warn_span!`], and
+//! [`error_span!`] exist as shorthand for constructing spans at various
+//! verbosity levels.
+//!
+//! ## Recording Span Creation
+//!
+//! The [`Attributes`] type contains data associated with a span, and is
+//! provided to the [`Subscriber`] when a new span is created. It contains
+//! the span's metadata, the ID of the span's parent if one was explicitly set,
+//! and any fields whose values were recorded when the span was constructed.
+//! The subscriber may then choose to cache the data for future use, record
+//! it in some manner, or discard it completely.
```

Comment:
This last sentence is pretty vague given that I (currently) have no idea what a subscriber is if I have followed the docs thus far. I think elaborating a bit here would be worthwhile.